packages/js/product-editor/src/hooks/use-product-variations-helper.ts

```diff
@@ -111,6 +113,15 @@ export function useProductVariationsHelper() {
 					invalidateResolutionForStoreSelector(
 						'getProductVariations'
 					);
+					if ( variations && variations.length > 0 ) {
```

Comment:
Well this does not work when there are old variations in cache, when invalidating the old variations the a request is made and the server responds with 404. Also we might need to invalidate the parent product request here too, since generate variations affect the `variations` property of the product which is used in the variation switcher.

Reply:
Hmm yeah I could do the `invalidateResolutionForStoreSelector( 'getEntityRecord' )` which would do it for all of them, but I was trying to avoid that.

Reply:
I know but if we don't do it then the variation ids from the parent product never get updated, so when we land to the single variation page, that list is being used for the swiper and it'll be outdated. Once way to avoid that is updating the local copy of the variation ids within the cached parent product right after we generate variations, which is going to force the user to save the product before leaving the page.

Reply:
I included the invalidation of the product here.